eng/common/pipelines/templates/steps/resolve-artifact-name.yml

```diff
@@ -0,0 +1,19 @@
+# Sets a variable that can be used later on to safely upload an artifact. This means that failed iterations of a build phase
+# will note REMAIN failing after the appropriate updates have been made to the source.
+# ArtifactNameVariable - The name of the DevOps Variable that will be consumed in the PublishPipelineArtifact steps
+# DefaultArtifactName  - The name of the artifact in the "successful" case.
+
+parameters:
+  ArtifactNameVariable: 'Publishing.Artifact.Name'
+  DefaultArtifactName: 'packaging'
+
+steps:
+  - pwsh: |
```

Comment:
@mitchdenny and I were chatting about this and one option we came up with that would accomplish the same thing while avoiding the variable name dancing would be to just have to publish calls one for success and one for failure. Something like:

```
  - task: PublishPipelineArtifact@1
    condition: succeeded()
    displayName: 'Publish ${{ parameters.ArtifactName }} Artifacts'
    inputs:
      artifactName: ${{ parameters.ArtifactName }}
      pathtoPublish: ${{ parameters.ArtifactPath }}
  - task: PublishPipelineArtifact@1
    condition: failed()
    displayName: 'Publish failed ${{ parameters.ArtifactName }} Artifacts'
    inputs:
      artifactName: '${{ parameters.ArtifactName }}-FailedAttempt$(System.JobAttempt)'
      pathtoPublish: ${{ parameters.ArtifactPath }}
```

That would allow us to have a generic publish-artifact.yml template we could use in all cases.

Reply:
Done. Testing.